src/common.jl

```diff
@@ -1,29 +1,31 @@
 const Cmode_t = Cushort
 
 "Generic structure used for passing keys and data in and out of the database."
-type MDBValue
+immutable MDBValue
     size::Csize_t   # size of the data item
     data::Ptr{Void} # address of the data item
 end
 MDBValue() = MDBValue(zero(Csize_t), C_NULL)
-function MDBValue(val)
-    val_size = sizeof(val)
-    val = isa(val, Number) ? typeof(val)[val] : val
-    return MDBValue(val_size, pointer(val))
-end
-
-function convert{T}(::Type{T}, mdb_val_ref::Ref{MDBValue})
-    mdb_val = mdb_val_ref[]
-    return if T <: String
-        unsafe_string(convert(Ptr{UInt8}, mdb_val.data), mdb_val.size)
-    elseif T <: AbstractVector
-        E = eltype(T)
-        nvals = floor(Int, mdb_val.size/sizeof(E))
-        unsafe_wrap(T, convert(Ptr{E}, mdb_val.data), nvals)
+MDBValue(_::Void) = MDBValue()
+MDBValue{T<:Union{String,Array}}(val::T) = MDBValue(sizeof(val), pointer(val))
+function MDBValue{T}(val::T)
+    if isbits(T)
```

Comment:
I still want a bit type value to be used. What if we create a field in `MDBValue` of value type to keep copy of the value and anchor it against GC sweeps.
```julia
struct MDBValue{T}
      size::Csize_t   # size of the data item
      value::T
      data::Ptr{Void} # address of the data item
end

function MDBValue(val::T) where {T <: Number}
    v =  MDBValue(sizeof(val), val, C_NULL)
    v.data = pointer(v.value)
    return v
end
```

Reply:
Setting bits type data is still possible in the current model.

This should work too... I tried to avoid this as the struct is passed into C (but should be okay as the additional data is at the end). I can give it a shot.

When `T <: Number`, `value` is `Vector{T}` and not `T`. So maybe `value::Union{T,Vector{T}}` is what we need.

Reply:
I think that `MDBVaule` should only work with arrays, strings and immutable structs. The bits types can be wrapped in the `put!` call into the array before constructing `MDBValue` object. Thus there will be no difference between bit type and 1-element array of such type.